### DISCUSSION TITLE
3.2.9

### DISCUSSION THREAD

**Discussion by mathiascode** [nicotine-plus/nicotine-plus#2432] [Releases]
WINDOWS USERS: The installer format has changed in Nicotine+ 3.2.0. If you are upgrading from Nicotine+ 3.1.1 or earlier, please uninstall Nicotine+ first (this will not remove your existing settings).

### Corrections

 * Reduced memory usage when browsing large shares
 * Fixed a crash on some systems after running for a few days
 * Fixed an issue where some private messages were ignored after a user reconnected
 * Fixed an issue where downloads with long file names could fail on eCryptfs file systems
 * Fixed an issue where the displayed total percentage of folder transfers was incorrect
 * Fixed an issue where the tray icon could disappear after locking the screen

### Issues closed on GitHub

 * Messages are wonky ([#2329](https://github.com/nicotine-plus/nicotine-plus/issues/2329))
 * Current download progress reflects single download instead of total ([#2373](https://github.com/nicotine-plus/nicotine-plus/issues/2373))
 * Download I/O error: \[Errno 36\] File name too long ([#2375](https://github.com/nicotine-plus/nicotine-plus/issues/2375))

<hr /><em>This discussion was created from the release <a href='https://github.com/nicotine-plus/nicotine-plus/releases/tag/3.2.9'>3.2.9</a>.</em>

**Comment by jcor29** [comment]
Hi,
I would like to know, if a character base version would be possible ?. 
I explain.  I use nicotine on my laptop whole day, as electricity is becoming an issue price wise, 
It could make  sence to be able to run on a raspberry in (character mode, curses  or something else)

**Reply by slook** [reply]
Yes, if you start [the development version of Nicotine+](https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md) 3.3.0.dev5 with the `--headless` switch then it has a basic command line interface. Type `help` for a list of available commands...

```
help
Listing 14 available commands:

Nicotine+ Commands:
	/help, /? [query]  -  List available commands
	/away, /a  -  Toggle away status
	/plugin <toggle|info> <plugin_name>  -  Manage plugins
	/quit, /q, /exit [-force]  -  Quit Nicotine+

Users:
	/add, /buddy <user>  -  Add user to buddy list
	/rem, /unbuddy <buddy>  -  Remove buddy from buddy list

Network Filters:
	/ip <user or ip>  -  Show IP address or username
	/ban <user or ip>  -  Block connections from user or IP address
	/unban <user or ip>  -  Remove user or IP address from ban lists

Shares:
	/share <public|buddy> <folder path>  -  Add share
	/unshare <virtual name or folder path>  -  Remove share
	/shares, /ls [public] [buddy]  -  List shares
	/rescan [force|rebuild]  -  Rescan shares

Port Checker:
	/port <user>  -  Check firewall state of user

Type /help [query] to list similar commands
```

Currently, only uploading and banning related functions are supported, although more features such as searching and downloading are possible in the future, but only if there is enough demand for it.

The best option for a fully fledged headless client at the present time is slskd https://github.com/slskd/slskd A modern client-server application for the Soulseek network.

> It could make sence to be able to run on a raspberry

Yes, many users do run those. Also on an Android with [Termux](https://f-droid.org/en/packages/com.termux) does work well if you have an old phone with a MicroSDXC slot or whatever.

> character mode, curses or something

Yes, I tried to port the [Mucous curses interface from the old Museek+ project](https://github.com/eLvErDe/museek-plus/tree/master/mucous/pymucous), a sibling of Nicotine+, but it is too specific and old (Python 2 only), so I concluded it will be necessary to start from scratch.

We could consider using something like the blessings or [blessed](https://github.com/jquast/blessed) library, which is a Python wrapper around curses that is small enough to be bundled inside the nicotine package (we like to avoid external dependencies) to make headless plugin development easier.

Alternatively there is https://github.com/nicotine-plus/nicotine-plus/discussions/2312 or [the slskd client](https://github.com/slskd/slskd) if you want a web based GUI.

**Comment by jcor29** [comment]
Hi,
Many thanks for your quick answer,
I will right away have a look at the slsdk client
Best Regards
Jacques

**Comment by jcor29** [comment]
Hi ,
Sorry for bothering you again,
I tried to get in running with docker on my raspberry,
but i'm getting some weird docker errors
Is there any way to install it without docker ?
Best Regards
Jacques

**Reply by slook** [reply]
Not afaik. See https://github.com/slskd/slskd/tree/master/docs and refer your questions to https://github.com/slskd/slskd/discussions

https://github.com/slskd/slskd/blob/master/docs/docker.md
